FHEM/59_Buienradar.pm

```diff
@@ -318,61 +387,51 @@ sub Get {
 }
 
 sub Attr {
-    my ($command, $device_name, $attribute_name, $attribute_value) = @_;
-    my $hash = $::defs{$device_name};
-
-    Debugging(
-        'Attr called', '\n',
-        Dumper (
-            $command, $device_name, $attribute_name, $attribute_value
-        )
-    );
+    my ($command, $name, $attribute_name, $attribute_value) = @_;
+    my $hash = GetHash($name);
+    
+    Debugging($name, Dumper({
+        command     =>  $command,
+        device      =>  $name,
+        attribute   =>  $attribute_name,
+        value       =>  $attribute_value
+    }));
 
     given ($attribute_name) {
         # JFTR: disabled will also set disable to be compatible to FHEM::IsDisabled()
         #       This is a ugly hack, with some side-effects like you can set disabled, disable will be automatically
         #       set, you can delete disable but disabled will still be set.
         when ('disabled') {
-            Debugging(
-                Dumper (
-                    {
-                        'attribute_value' => $attribute_value,
-                        'attr' => 'disabled',
-                        'command' => $command,
-                    }
-                )
-            );
-
             given ($command) {
                 when ('set') {
                     return qq[${attribute_value} is not a valid value for disabled. Only 'on' or 'off' are allowed!]
-                        if $attribute_value !~ /^(on|off|0|1)$/;
+                        if $attribute_value !~ /^(?: on | off | 0 | 1 )$/x;
 
-                    if ($attribute_value =~ /(on|1)/) {
-                        ::RemoveInternalTimer( $hash, 'FHEM::Buienradar::Timer' );
-                        $::attr{$device_name}{'disable'} = 1;
+                    if ($attribute_value =~ /(?: on | 1)/x) {
+                        ::RemoveInternalTimer( $hash,\&FHEM::Buienradar::Timer );
+                        Disable($name);
                         $hash->{NEXTUPDATE} = undef;
                         $hash->{STATE} = 'inactive';
                         return;
                     }
 
-                    if ($attribute_value =~ /(off|0)/) {
-                        $::attr{$device_name}{'disable'} = 0;
+                    if ($attribute_value =~ /(off|0)/x) {
```

Comment:
Sicher, dass off irgendwo im Wert vorkommen kann / soll? Zeile 408 habe ich so verstanden, dass es mit on , off 0 oder 1 sein muss. Dann macht das hier mit off bzw 0 außer am Anfang keinen Sinn.
`=~ /^(?:off|0)$/x) {`

Reply:
In [L408](https://github.com/fhem/mod-Buienradar/blob/1f30d4ab3d20e3782e504ce23355afaf8e234a0a/FHEM/59_Buienradar.pm#L408) wird geprüft ob `on,off,0,1` **nicht** vorkommen, dann wird `Attr()` mit einem Fehler verlassen. Der Block zwischen L410 und L422 unterscheidet den Fall, dass jemand `disable` auf `on` oder `off` (mit Behandlung des unschönen `1` bzw. `0`) setzt und nicht nur einfach löscht.

Reply:
Aber du hast mich darauf gebracht, dass ich da einen Regexp vergessen habe: #108